### PR TITLE
feat(sdk-coin-sol): add sol close ATA recovery support along with tok…

### PR DESCRIPTION
**Pull request by DinshawKothari**
…en recovery

Ticket: WIN-2648

### CODE REVIEW

File: modules/sdk-coin-sol/src/sol.ts [hunk 1]
```diff
@@ -1043,15 +1041,63 @@ export class Sol extends BaseCoin {
...
+      txBuilder = factory
+        .getTokenTransferBuilder()
+        .nonce(blockhash)
+        .sender(bs58EncodedPublicKey)
+        .fee({ amount: feePerSignature })
+        .associatedTokenAccountRent(rentExemptAmount.toString())
+        .feePayer(bs58EncodedPublicKey);
+
+      const network = this.getNetwork();
+      const token = getSolTokenFromAddress(tokenInfo.info.mint, network);
```
Comment (mukeshsp) [inline modules/sdk-coin-sol/src/sol.ts:1077]: what if there are multiple tokens ?

File: modules/sdk-coin-sol/src/sol.ts [hunk 2]
```diff
@@ -1043,15 +1041,63 @@ export class Sol extends BaseCoin {
...
+    const tokenInfo = await this.getTokenAccountInfo(params.closeAtaAddress);
+    const tokenBalance = Number(tokenInfo.info.tokenAmount.amount);
+
+    if (tokenBalance > 0) {
+      // closeATA address has some token balance, it needs to be withdrawn before closing ATA
+      console.log(
+        `closeATA address ${params.closeAtaAddress} has token balance ${tokenBalance}, it needs to be withdrawn before closing ATA address`
+      );
+
+      if (!params.recoveryDestinationAtaAddress || !this.isValidAddress(params.recoveryDestinationAtaAddress)) {
```
Comment (mukeshsp) [inline modules/sdk-coin-sol/src/sol.ts:1062]: FYI : recoveryDestinationAtaAddress should be belong to same token account

File: modules/sdk-coin-sol/src/sol.ts (hunk 1 above)
Comment (DinshawKothari) [reply modules/sdk-coin-sol/src/sol.ts:1077]: There won't be multiple tokens, we are not dealing with receive address here, but we are only interested in closeATA address, and a ATA address will always belong to just one token.

File: modules/sdk-coin-sol/src/sol.ts (hunk 2 above)
Comment (DinshawKothari) [reply modules/sdk-coin-sol/src/sol.ts:1062]: Yes correct, we will provide this detail in coin sandbox script clearly.

**Review by mukeshsp — APPROVED** [review]

**Pull request merged**